docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,137 @@
+  should be avoided
+
+- Dynamically determining a given release's RHEL version. The RHEL version will
+  be a part of configuration tied to the minor release
+
+## Proposal
+
+Two additional containers will run alongside assisted-service in the same pod.
+These "installer-runner" containers will, when triggered by assisted-service,
+run a specified installer binary to generate the manifests required for a
+particular install. These manifests will then be uploaded to whatever storage
+is in use for this deployment (local for on-prem, or s3 for SaaS) and
+assisted-service will take over as usual from there.
+
+Each of these containers should maintain its own cache of installers as
+assisted-service does now and should be built with the required packages to run
+the installer in FIPS compliance mode.
+
+### User Stories
+
+#### Story 1
+
+As a Cluster Creator, I want to install FIPS compliant clusters for any supported OpenShift version
+
+### Implementation Details/Notes/Constraints
+
+#### New Images
+
+Two new container images will need to be built, and published for every release
+footprint we support. These images will be created based on existing
+assisted-service code, but could be split into their own independent projects
+later.
+
+### Risks and Mitigations
+
+Shipping a new image is a non-trivial process. This may take more time than we
+have to set up. We could likely get away with using the existing
+assisted-service image with a different entrypoint for one of the runner images,
+but that still requires us to publish a new one for the architecture
+assisted-service will not be using.
+
+## Design Details [optional]
+
+The new runner containers will expose a HTTP server listening only for local
+connections. assisted-service will POST to this server when it needs manifests
+generated. The runner container will respond with any error that occurred while
+generating the manifests.
+
+### Open Questions
+
+1. What does the API look like for the runner containers? What data should be
+   passed in an API call and what should be configured in the container
+   environment?
+
+### UI Impact
+
+No need to go into great detail about the UI changes that will need to be made.
+However, this is an excellent time to mention 1) if UI changes are required if
+this enhancement were accepted and 2) at a high-level what those UI changes
+would be.
+
+### Test Plan
+
+FIPS and regular installs should be tested for all supported OpenShift versions.
+Since this should be mostly transparent to the user, regression testing in
+addition to testing 4.16 with and without FIPS should be sufficient.
+
+## Drawbacks
+
+This is a complicated change in architecture something simpler might be more
+desirable.
+
+Creating two additional containers in a pod makes the assisted service more
+expensive to scale.
+
+Creating, maintaining, and releasing additional images is a non-trivial amount
+of additional work.
+
+## Alternatives
```

Comment:
if we're saying that <4.16 is not fips complaint anyhow, can we say that for those releases we gonna use the statically linked one and for >=4.16, dynamically linked with rhel9? Can we use single existing container then?

Reply:
I don't think we're saying that. OpenShift can be installed in FIPS mode today, and that's been the case for a while.

Reply:
in ACM/SaaS world? or in general?

Reply:
In general and in ACM IIUC

Reply:
Sources for "OpenShift can be installed in FIPS mode today" include [4.12 docs for FIPS installation][1].  And [ROSA FedRAMP docs][2]:

> ROSA only uses FIPS-validated modules to process cryptographic libraries.

[1]: https://docs.openshift.com/container-platform/4.12/installing/installing-fips.html
[2]: https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/govcloud-rosa.html